### PR TITLE
Added a super provider to access both chains

### LINKED ISSUE #24: Introduce Mosaic abstraction
Add a chain-aware web3 superprovider (provides multiple web3 providers), like we had it at https://github.com/OpenSTFoundation/mosaic.js/tree/c7e31011613c2f4ef2a70cc0b0e3a966193bbe29#create-mosiac-object, as we had it before merging in `mosaic-tbd`, and complementatry to the `ChainSetup` (which is focused on chain-aware deployment).

1:1 Origin:Auxiliary

### PR DESCRIPTION
**Pull request by schemar**
I wasn't sure which parameters we need on the class, so please check.
I wasn't sure where to put the tests. Please check.

Fixes #24

### CODE REVIEW

**Comment by schemar** [comment]
Also, maybe you can come up with better names.
And *also:* should we have a separate `Chain` class?

File: src/MosaicChains.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+'use strict';
+
+const Chain = require('./Chain');
+
+/**
+ * MosaicChains stores access to two chains of a mosaic set-up: origin and auxiliary.
+ * @param {Chain} origin The origin chain of this mosaic set-up.
+ * @param {Chain} auxiliary The auxiliary chain of this mosaic set-up.
+ */
+class MosaicChains {
```
Comment (hobofan) [inline src/MosaicChains.js]: I rather liked the shorter `Mosaic` naming, but also not too strong of a opinion on that 🤷‍♀️

File: src/Chain.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+'use strict';
+
+const Web3 = require('web3');
+
+/**
+ * Chain stores a web3 instance to access a chain as well as the addresses of the required
+ * contracts.
+ * @param {Web3} web3 A web3 instance with a provider to access this chain.
+ * @param {Object} contractAddresses Contract addresses, indexed by identifier, e.g. Anchor or
```
Comment (hobofan) [inline src/Chain.js:29]: > Anchor or

or what?

**Review by hobofan — CHANGES_REQUESTED** [review]
`Chain` and `ChainSetup` should be exported in `index.js`. Otherwise looks good!

File: src/Chain.js (hunk 2 above)
Comment (schemar) [reply src/Chain.js:29]: Thanks fixed

File: src/MosaicChains.js (hunk 1 above)
Comment (schemar) [reply src/MosaicChains.js]: Renamed

**Review by hobofan — APPROVED** [review]

**Pull request merged**